ring_buffer/ring_buffer.py

```diff
@@ -8,16 +8,49 @@ def __init__(self, capacity):
         self.storage = DoublyLinkedList()
 
     def append(self, item):
```

Comment:
Really solid implementation here; you really had a clear plan and used the pointers effectively in a way that not everybody managed, good work!